src/Response.php

```diff
@@ -90,17 +90,13 @@ class Response implements ResponseInterface
      * @param string|null                          $reason  Reason phrase (when empty a default will be used based on the status code)
      */
     public function __construct(
-        $status = 200,
+        int $status = 200,
         array $headers = [],
         $body = null,
-        $version = '1.1',
-        $reason = null
+        string $version = '1.1',
+        string $reason = null
     ) {
-        if (filter_var($status, FILTER_VALIDATE_INT) === false) {
```

Comment:
It feels so great removing this.

Reply:
It will be added back... :(

https://github.com/guzzle/psr7/pull/250/files#diff-f89246dce7e387482b2c6c584a5e7e9dR144

The PR above targets 1.x version... Unfortunately PSR-7 does not use PHP7 so values passed through `with* ` must be validated... 

I believe we must merge the above one first that fixes some stuff according to PSR and then proceed with this one.. unless if you want to leave 1.x behind and move to a new era of 2.x and everything should target 2.x

Reply:
Also we must note the following... the different behavior that it will be cause from:

```
new Response(null); // Throws TypeError
````
vs

```
$r = new Response();
$r->withStatusCode($statusCode); //throws \InvalidArgumentException
```

So just thinking out loud... is typehints to public functions something that can be used?

Reply:
We need to validate the `with*` method as we cannot add type declarations there. But the constructor can add type declaration and thus also does not need to validate.